src/main/java/seedu/manager/parser/Parser.java

```diff
@@ -145,18 +156,37 @@ public class Parser {
-    private static final String REMOVE_PARTICIPANT_REGEX = "(-p|-e)";
-    private static final String MARK_EVENT_REGEX = "-e|-s";
-    private static final String MARK_PARTICIPANT_REGEX = "-p|-e|-s";
-    private static final String FIND_REGEX = "\\s*(-e|-p)\\s*";
-    private static final String VIEW_REGEX = "(-e|-y)";
-    private static final String MARK_ITEM_REGEX = "-m|-e|-s";
+    private static final String EVENT_FLAG_REGEX = "(-e|-t|-v|-u)";
+    private static final String EDIT_EVENT_ATTRIBUTE_FLAG_REGEX = "(-e|-name|-t|-v|-u)";
+    private static final String PARTICIPANT_FLAG_REGEX = "(-p|-n|-email|-e)";
+    private static final String REMOVE_PARTICIPANT_FLAG_REGEX = "(-p|-e)";
+    private static final String ITEM_FLAG_REGEX = "(-m|-e)";
+    private static final String MARK_EVENT_FLAG_REGEX = "-e|-s";
+    private static final String MARK_PARTICIPANT_FLAG_REGEX = "-p|-e|-s";
+    private static final String FIND_FLAG_REGEX = "\\s*(-e|-p)\\s*";
+    private static final String VIEW_FLAG_REGEX = "(-e|-y)";
+    private static final String MARK_ITEM_FLAG_REGEX = "-m|-e|-s";
+    private static final String REMOVE_EVENT_FLAG_REGEX = "-e";
+
+    private static final String ADD_EVENT_REGEX = "add\\s+-e\\s+(.*?)\\s+-t\\s+(.*?)\\s+-v\\s+(.*?)\\s+-u\\s+(.*)";
+    private static final String EDIT_EVENT_ATTRIBUTE_REGEX = "edit\\s+-e\\s+(.*?)\\s+" +
+            "-name\\s+(.*?)\\s+-t\\s+(.*?)\\s+-v\\s+(.*?)\\s+-u\\s+(.*)";
+    private static final String ADD_PARTICIPANT_REGEX = "add\\s+-p\\s+(.*?)\\s+" +
+            "-n\\s+(.*?)\\s+-email\\s+(.*?)\\s+-e\\s+(.*)";
+    private static final String EDIT_PARTICIPANT_REGEX = "edit\\s+-p\\s+(.*?)\\s+" +
+            "-n\\s+(.*?)\\s+-email\\s+(.*?)\\s+-e\\s+(.*)";
+    private static final String ADD_ITEM_REGEX = "add\\s+-m\\s+(.*?)\\s+-e\\s+(.*)";
+    private static final String REMOVE_ITEM_REGEX = "remove\\s+-m\\s+(.*?)\\s+-e\\s+(.*)";
+    private static final String EDIT_ITEM_REGEX = "edit\\s+-m\\s+(.*?)\\s+-e\\s+(.*)";
+    private static final String REMOVE_PARTICIPANT_REGEX = "remove\\s+-p\\s+(.*?)\\s+-e\\s+(.*)";
+    private static final String MARK_EVENT_REGEX = "mark\\s+-e\\s+(.*?)\\s+-s\\s+(.*)";
+    private static final String MARK_PARTICIPANT_REGEX = "mark\\s+-p\\s+(.*?)\\s+-e\\s+(.*?)\\s+-s\\s+(.*)";
+    private static final String FIND_REGEX = "find\\s+-e\\s+(.*?)\\s+-p\\s+(.*)";
+    private static final String VIEW_REGEX = "view\\s+-e\\s+(.*?)\\s+-y\\s+(.*)";
+    private static final String MARK_ITEM_REGEX = "mark\\s+-m\\s+(.*?)\\s+-e\\s+(.*?)\\s+-s\\s+(.*)";
+    private static final String REMOVE_EVENT_REGEX = "remove\\s+-e\\s+(.*)";
     private static final Pattern PHONE_NUMBER_PATTERN = Pattern.compile("\\d{8}");
-    private static final Pattern EMAIL_PATTERN = Pattern.compile("^[A-Za-z0-9+_.-]+@[A-Za-z0-9-]+\\.[A-Za-z0-9-]+$");
+    private static final Pattern EMAIL_PATTERN = Pattern.compile("^[A-Za-z0-9+_.-]+@[A-Za-z0-9-]+\\.[A-Za-z0-9.-]+$");
```

Comment:
^[A-Za-z0-9+_.-]+@[A-Za-z0-9-]+(\\.[A-Za-z0-9-]+)+$

can change this regex to this instead as the current one allows ending with .